stac_fastapi/pgstac/core.py

```diff
@@ -184,12 +184,11 @@ async def _search_base(  # noqa: C901
         prev: Optional[str] = items.pop("prev", None)
         collection = ItemCollection(**items)
 
-        exclude = search_request.fields.exclude
-        if exclude and len(exclude) == 0:
-            exclude = None
-        include = search_request.fields.include
-        if include and len(include) == 0:
-            include = None
+        include: Set[str] = set()
+        exclude: Set[str] = set()
+        if fields := getattr(search_request, "fields", None):
+            include = fields.include or set()
+            exclude = fields.exclude or set()
```

Comment:
fields = getattr(search_request, "fields", None)
include: Set[str] = fields.include if fields and fields.include else set()
exclude: Set[str] = fields.exclude if fields and fields.exclude else set()

A little shorter maybe